Description: Deal with hex conversion in i386
 Use format to avoid L suffix appended by hex()
 function on 32 bit architectures to long
Author: James Page <james.page@ubuntu.com>
Forwarded: https://review.openstack.org/229819
Bug: https://bugs.launchpad.net/neutron/+bug/1501703

--- a/neutron/agent/linux/ip_lib.py
+++ b/neutron/agent/linux/ip_lib.py
@@ -331,7 +331,7 @@ class IpRuleCommand(IpCommandBase):
                 integers.append(0xffffffff)
 
             # We now have two integers in a list.  Convert to canonical string.
-            return '/'.join(map(hex, integers))
+            return '{0:#x}/{1:#x}'.format(*integers)
 
         def canonicalize(item):
             k, v = item
